cep-XXXX.md

```diff
@@ -0,0 +1,424 @@
+
+- `<OCI-compatible channel path>`: `^[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*(\/[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*)*$` (same as the regex for an OCI repository `<name>`)
+- `<OCI-compatible label>`: `^[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*(\/[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*)*$` (same as the regex for an OCI repository `<name>`)
+- `<OCI-compatible subdir>`: `^[a-z0-9]+(-[a-z0-9]+)*$`
+
+### Subdir Repository Namespace Designations
+
+The namespace after the combination `<OCI conda channel base URL>/<OCI-compatible subdir>/` has the following designations
+(where the `*` wildcard indicates any valid OCI repository `<name>` component or part of a component as allowed by
+the OCI repository `<name>` regex):
+
+- `<OCI conda channel base URL>/<OCI-compatible subdir>/r*`: reserved for repodata JSON artifacts, run exports JSON artifacts, and
+  future CEP-defined extensions for sharded repodata (defined in [CEP 16][cep16]).
+- `<OCI conda channel base URL>/<OCI-compatible subdir>/c*`: reserved for OCI-encoded conda package names (The character `c`
+  is prepended to the conda package name to indicate that it is an OCI-encoded conda package name as described below.)
+- `<OCI conda channel base URL>/<OCI-compatible subdir>/h*`: reserved for hashed OCI-encoded conda package names
+- `<OCI conda channel base URL>/<OCI-compatible subdir>/p*`: reserved for repodata patch instructions
+
+These designations are used to prevent namespace collisions between the various components of an OCI conda channel.
+
+### Conda Artifact OCI Repository Names
+
+Conda artifacts MUST be stored in the OCI repository of an OCI conda channel under their `<OCI-compatible subdir>` as the OCI repository
+
+```text
+<OCI conda channel base URL>/<OCI-compatible subdir>/<OCI-encoded package name>
+```
+
+The contents and mediaTypes of OCI blobs that make up the conda package artifact, along with how to compute the OCI tag, are defined below as well.
+
+The `<OCI-encoded package name>` MUST be computed according to the following rules:
+
+- All package names MUST be prepended by the character `c`.
+- If the combined string `<OCI conda channel base URL>/<OCI-compatible subdir>/<OCI-encoded package name>` exceeds 128 characters
+  in length, the`<OCI-encoded package name>` MUST be replaced by the SHA256 hash of the `<OCI-encoded package name>`
+  in hexadecimal form as `h<SHA256 in hexadecimal>`. The letter `h` is reserved to indicate a hashed OCI-encoded package name as described above.
+
+`<OCI-encoded package name>`s that are not hashed can be decoded by reversing the encoding rules above.
+
+For example, the package name `_libgcc_mutex` is encoded to OCI-form as `c_libgcc_mutex`, and the package name `c_libgcc_mutex`
+is encoded to OCI-form as `cc_zlibgcc_mutex`. A very long package name like `gblah000...000` would be encoded to its OCI-form
+`cgblah000...000`, then hashed via SHA256, yielding something like `h<hexdigest of SHA256 hash>`.
```

Comment:
Let's add a note here saying that the original components can be found as annotation metadata, as discussed below.

Reply:
So to be clear, I didn't end up hashing everything like we discussed. It is just the package name that is hashed. The rationale here is two-fold

1. Packages move between channels and so it didn't make sense to me to hash the channel and label with the package name. 

2. The hashes would occupy a specific namespace in the space of channels. While collisions themselves through normal operations would be rare, I was concerned that there might be security implications for this down the road.